transcom/mymove
# B-17477 Revert some changes in ShipmentWeightDetails to fix bug

## [B-17477](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a842007)

## Summary

In my [previous PR related to this story](https://github.com/transcom/mymove/pull/11540), I mistakenly used the value calculatedBillableWeight as the Actual shipment weight, misunderstanding the use of the word billable. Upon a further inspection prior to PO Acceptance demo this morning, I realized that calculatedBillableWeight would in some cases return mto_shipments:billable_weight_cap. After running it by DS, we determined the logic I had removed from the front end to determine the Actual shipment weight should be returned to the component.

To that end I removed the calculatedBillableWeight prop being passed to ShipmentWeightDetails and updated ShipmentWeightDetails to use the previous logic to determine the Actual shipment weight (returnLowestValue). I also updated the prop names to the more semantic initialWeight in both the component and the test files.

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

(copied from my previous PR)
1. Login as a new or existing customer and create a new HHG shipment
2. Login as a services counselor, access the newly created HHG, and fill out necessary information, then click Submit move details
3. Login as a TOO, access the new move; select the move and approve the Move management service item. Confirm that all weight fields in the shipment appear and don't have values
4. Login in a prime simulator role and access the new move; Update the shipment and include values for estimated weight and actual weight
5. Login as a TOO, access the move; confirm estimated weight, initial weight, and actual shipment weight are populated with values. Request a reweigh
6. Login in a prime simulator role and access the move; respond to the reweigh request and enter a number
7. Login as a TOO, access the move, and confirm that a reweigh value appears, and the Actual shipment weight displays the lower of the two values between Initial weight and Reweigh weight
